src/messages.ts

```diff
@@ -548,14 +548,43 @@ export class Messages<T extends string> {
     exitCodeOrCause?: number | Error,
     cause?: Error
   ): SfError {
-    const structuredMessage = this.formatMessageContents('error', key, tokens, actionTokens);
-    return new SfError(
-      structuredMessage.message,
-      structuredMessage.name,
-      structuredMessage.actions,
-      exitCodeOrCause,
-      cause
-    );
+    const { message, name, actions } = this.formatMessageContents({
+      type: 'error',
+      key,
+      tokens,
+      actionTokens,
+    });
+    return new SfError(message, name, actions, exitCodeOrCause, cause);
+  }
+
+  /**
+   * SfError wants error to end with the word Error.  Use this to create errors while preserving their existing name (for compatibility reasons).
+   *
+   * @deprecated Use `createError` instead unless you need to preserver the error name to avoid breaking changes.
+   * `error.name` will be the upper-cased key, remove prefixed `error.` and will always end in Error.
+   * `error.actions` will be loaded using `${key}.actions` if available.
+   *
+   * @param key The key of the error message.
+   * @param tokens The error message tokens.
+   * @param actionTokens The action messages tokens.
+   * @param exitCodeOrCause The exit code which will be used by SfdxCommand or the underlying error that caused this error to be raised.
+   * @param cause The underlying error that caused this error to be raised.
+   */
+  public createErrorButPreserveName(
```

Comment:
I came here initially to show my appreciate for intention revealing function names 😄.